modules/managers/galasa-managers-parent/galasa-managers-imstm-parent/dev.galasa.imstm.manager/src/main/java/dev/galasa/imstm/internal/ImstmDefaultLogonProvider.java

```diff
@@ -0,0 +1,133 @@
+        } catch (ImstmManagerException e) {
+            throw new ImstmManagerException("Problem retrieving logon text for the default logon provider", e);
+        }
+    }
+
+    @Override
+    public boolean logonToImsSystem(IImsTerminal imsTerminal) throws ImstmManagerException {
+
+        try {
+            if (!imsTerminal.isConnected()) {
+                imsTerminal.connect();
+            }
+
+            // Ensure we can type something first
+            imsTerminal.waitForKeyboard();
+
+            // Check we are at the right screen
+            if (initialText != null) {
+                checkForInitialText(imsTerminal);
+            }
+
+            IImsSystem system = imsTerminal.getImsSystem();
+            imsTerminal.type(system.getZosImage().getVtamLogonString(system.getApplid())).enter().wfk();
+
+            waitForSignonScreen(imsTerminal);
+            logger.debug("Logged onto " + system);
+
+            if (imsTerminal.getLoginCredentialsTag().isEmpty()) {
+                throw new ImstmManagerException("No login credentials provided");
+            } else {
+                ICredentialsUsername creds = (ICredentialsUsername) this.cs.getCredentials(imsTerminal.getLoginCredentialsTag());
+
+                imsTerminal.positionCursorToFieldContaining("USERID:");
+                imsTerminal.tab();
+                imsTerminal.type(creds.getUsername());
+                if (creds instanceof ICredentialsUsernamePassword) {
+                    String pw = ((ICredentialsUsernamePassword) creds).getPassword();
+                    cts.registerText(pw, "Password for credential tag: " + imsTerminal.getLoginCredentialsTag());
+                    imsTerminal.positionCursorToFieldContaining("PASSWORD:");
+                    imsTerminal.tab();
+                    imsTerminal.type(pw);
+                    }
+                imsTerminal.enter().wfk();
+
+                waitForSignedOnText(imsTerminal);
+                logger.debug("Logged into IMS TM as user: " + creds.getUsername());
+            }
+
+            imsTerminal.clear().wfk();
+
+        } catch (Zos3270Exception | CredentialsException e) {
+            throw new ImstmManagerException("Problem logging onto the IMS system", e);
+        }
+
+        return true;
+    }
+
+    private void checkForInitialText(IImsTerminal imsTerminal) throws ImstmManagerException {
+        try {
+            imsTerminal.waitForTextInField(initialText);
+        } catch (Exception e) {
+            throw new ImstmManagerException(
+                    "Unable to logon to IMS, initial screen does not contain '" + initialText + "'");
+        }
+    }
+
+    private void waitForSignonScreen(IImsTerminal imsTerminal) throws ImstmManagerException {
+        try {
+            imsTerminal.waitForTextInField("DFS3649A");
+        } catch (Exception e) {
+            throw new ImstmManagerException("Unable to wait for the initial IMS screen, looking for 'DFS3649I'",
```

Comment:
The code waits for "DFS3649A" and if not found says it's looking for 'DFS3649I' ? Use a constant and reference it twice ?

Reply:
Surprised this isn't configurable too, but I know little about IMS